checkstyle/src/main.rs

```diff
@@ -39,7 +43,7 @@ fn main() -> Result<(), std::io::Error> {
         );
     }
 
-    if !nonmatching_files.is_empty() {
+    if !nonmatching_files.is_empty() || !end_empty_line {
```

Comment:
```suggestion
    if !nonmatching_files.is_empty() || !end_empty_line.is_empty() {
```
and then we can go on to print out all of the files that don't match the regex, and all of the files that don't have empty ending lines. Remember to handle the edge cases where one of these arrays is empty but the other is not.